misc/python/materialize/version_list.py

```diff
@@ -148,7 +148,13 @@ def _resolve_image_tag_of_previous_release(
         )
 
     def _resolve_image_tag_of_latest_release(self, context: str) -> tuple[str, str]:
-        latest_published_version = get_latest_published_version()
+        # Even though we are on main we might be in an older state, pick the
+        # latest release that was before our current version.
+        current_version = MzVersion.parse_cargo()
```

Comment:
Should this only be done if we are on a `-dev` version?

Reply:
I think it's always better, otherwise if we rerun on an older branch we might try an upgrade from a newer version, which will fail.

Reply:
Ok, valid point. But should we go back one version (`get_previous_published_version`) in this case as well? I don't think so.

Reply:
I see two cases:
- You run this on `v0.110.0-dev`, line 386 will set the `version = MzVersion.create(version.major, version.minor, 0)` to `v0.110.0`, so we will use `v0.109.X` (where X is the latest). If we didn't go back one version we would do the same, because `v0.110.0-dev < v0.110.0`.
- You run this on `v0.110.5` with `previous_minor=True`, we'd test upgrade from `v0.109.X` (Interesting!), while with `previous_minor=False` we'd only test against `v0.110.4`, which probably has barely any changes. So if someone ignores the run on `v0.110.0` (which often is broken), then we'd never see the regressions vs `v0.109.X`

Reply:
Line 386 is in `get_previous_mz_version()` but we were using `get_latest_published_version()` before. But I guess `v0.109.X` will (accidentally) be the latest available version because `v0.110.x` won't exist yet? Still a bit confusing to assign `get_previous_published_version` to `latest_published_version`.

Reply:
Changed the name

Reply:
I am still not that happy because it conflicts with the function name. The function says we return the latest release but we don't.
What is the problem we are trying to solve?

Reply:
The latest release is usually wrong imo. We want the previous release from current state. I can't imagine a scenario where you need the latest release. I renamed the function.